ospatch/updates_windows.go

```diff
@@ -28,21 +28,46 @@ import (
 
 // SystemRebootRequired checks whether a system reboot is required.
 func SystemRebootRequired() (bool, error) {
-	reg := `SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired`
-	regPath := `HKLM:\` + reg
-	logger.Debugf("Checking if reboot required by looking at %s", regPath)
-	k, err := registry.OpenKey(registry.LOCAL_MACHINE, reg, registry.QUERY_VALUE)
+	// https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexw#remarks
+	logger.Debugf("Checking for PendingFileRenameOperations")
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SYSTEM\CurrentControlSet\Control\Session Manager`, registry.QUERY_VALUE)
 	if err != nil {
-		if err == registry.ErrNotExist {
-			logger.Debugf("%s does not exist, indicating no reboot is required.", regPath)
-			return false, nil
+		if err != registry.ErrNotExist {
+			return false, err
+		}
+	} else {
+		val, _, err := k.GetStringsValue("PendingFileRenameOperations")
+		if err != nil && err != registry.ErrNotExist {
+			return false, err
+		}
+		k.Close()
```

Comment:
below you only do this when `err == nil`. Is it ok to call this if there was an error that didn't trigger an early return?

Reply:
this would probably be an issue, changed to match other sections